### PR TITLE
Saas 17.4 crawlers once again awa

### PR DESCRIPTION
**Pull request by awa-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_event/controllers/main.py [hunk 1]
```diff
@@ -42,10 +42,14 @@ def _get_events_search_options(self, **post):
...
     def events(self, page=1, **searches):
-        if searches.get('tags') and request.httprequest.method == 'GET' and not searches.get('prevent_redirect'):
+        if len(searches.get('tags', '[]')) > 3 and request.httprequest.method == 'GET' and not searches.get('prevent_redirect'):
             # Previously, the tags were searched using GET, which caused issues with crawlers (too many hits)
             # We replaced those with POST to avoid that, but it's not sufficient as bots "remember" crawled pages for a while
             # This permanent redirect is placed to instruct the bots that this page is no longer valid
+            # Note: We allow a single tag to be GET, to keep crawlers & indexes on those pages
+            # What we really want to avoid is combinatorial explosions
+            # As 'tags' contains an array repr as a string, we need to test that it's greater than 3 (str length)
+            # '[]' is allowed, '[5]' is allowed, '[5,6]' is not
```
Comment (KangOl) [inline addons/website_event/controllers/main.py:52]: And `[42]`?

File: addons/website_event/controllers/main.py [hunk 2]
```diff
@@ -42,10 +42,14 @@ def _get_events_search_options(self, **post):
 
     @http.route(['/event', '/event/page/<int:page>', '/events', '/events/page/<int:page>'], type='http', auth="public", website=True, sitemap=sitemap_event)
     def events(self, page=1, **searches):
-        if searches.get('tags') and request.httprequest.method == 'GET' and not searches.get('prevent_redirect'):
+        if len(searches.get('tags', '[]')) > 3 and request.httprequest.method == 'GET' and not searches.get('prevent_redirect'):
```
Comment (KangOl) [inline addons/website_event/controllers/main.py:45]: Search for number of commas
```suggestion
        if searches.get('tags', '[]').count(',') > 0 and request.httprequest.method == 'GET' and not searches.get('prevent_redirect'):
```

File: addons/website_event/controllers/main.py (hunk 1 above)
Comment (awa-odoo) [reply addons/website_event/controllers/main.py]: Waw... good job me, I really need some vacation.

Here is a cat to make you forget this ever happened:
image

Thanks for the heads-up and the suggestion ^^

**Comment by awa-odoo** [comment]
Hello @JKE-be 

Does that look like what we need? Following our discussion from last week.
Tested quickly and looks like it works.

**Review by JKE-be — APPROVED** [review]
Thank you for the followup !

**Comment by JKE-be** [comment]
@robodoo rebase-ff

**Comment by tde-banana-odoo** [comment]
@robodoo r+

Thanks for the followup and quick update !

**Pull request closed without merging**